### PR TITLE
openthread: kconfig: Add experimental PSA Crypto support

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
Introduce PSA Crypto API in OpenThread.

- Enabled PSA internal storage
- Enabled OPENTHREAD_CRYPTO_PSA
- Enabled all PSA_WANT configs which are required by OpenThread
- Enabled PSA Crypto support in Matter

Known limitations and issues:

- Zephyr's settings are not initialized before the key manager initialization within OpenThread. Network key's storing should be done in the `AfterInit` method instead of the KeyManager class constructor - To be Implemented.

Workaround for testing: add the following lines to the `ext/oberon/psa/core/library/psa_crypto.c` file: 

include at the top of the file:

```
#include <zephyr/settings/settings.h>
```
to the psa_crypto_inti(void), after calling the psa_driver_wrapper_init() method:

```
settings_subsys_init();
```

### CODE REVIEW

**Review by canisLupus1313 — APPROVED** [review]

**Comment by ArekBalysNordic** [comment]
@SebastianBoe @frkv @Vge0rge Could you please look at the configuration regarding all PSA crypto dependencies that I set within the OpenThread library and verify that everything is consistent and set properly? Basically, we would like to start switching from MbedTLS to PSA Crypto API in Thread and Matter stacks and this PR is the very first attempt to start working on that. Finally, we would like to have a configuration that allows NRF devices to use PSA Crypto API without using MbedTLS at all. For now, we prepared an experimental kconfig that allows us to compile the Therad/Matter example with PSA crypto enabled, but there is probably the possibility to improve it or optimize it. I will appreciate all comments and suggestions.

**Comment by frkv** [comment]
> * Zephyr's settings are not initialized before the key manager initialization within OpenThread. Network key's storing should be done in the `AfterInit` method instead of the KeyManager class constructor - To be Implemented.

On first read-through it looks consistent with how we expect crypto features to be configured!

File: subsys/net/openthread/Kconfig [hunk 1]
```diff
@@ -19,12 +19,40 @@ config NORDIC_SECURITY_PROMPTLESS
...
 
 endchoice
 
+config OPENTHREAD_NRF_SECURITY_PSA
+	bool
+	select EXPERIMENTAL
+	select NRF_SECURITY
+	select MBEDTLS_PSA_CRYPTO_STORAGE_C
+	select MBEDTLS_ENABLE_HEAP
+	select PSA_NATIVE_ITS
```
[the remark this answers is not in the document]
Comment (ArekBalysNordic) [reply subsys/net/openthread/Kconfig:38]: Right, thanks!

**Comment by ArekBalysNordic** [comment]
Closing since all useful commits have been moved to https://github.com/nrfconnect/sdk-nrf/pull/12502. Adding support for PSA in Matter will be done in the other PR.

**Pull request closed without merging**